### PR TITLE
refactor: feature based folder organization

### PR DESCRIPTION
**Pull request by neurosnap**
The goal with this refactor is to keep things related to an entity (e.g.
asset, organization, environment, etc) colocated with each other.  This
has a few benefits:

- when refactoring an entity, all the code is located next to each other
- there is a home for all things related to an entity

I've also tried to flatten the folder structure so there is less
nesting.  I also tried to remove any possible naming collisions.  It's
very common for us to have a variable in our functions `assets` and
further we iterate over them and have an `asset` variable name.  This
often collides with any packages that we've named `asset` or `assets`.
Now we have a `libasset` package that is easy to autocomplete + import.

I also removed the `common` package collision by only having
`ui/common`.

We will want to continue to experiment with a feature based folder
organization, but I think this helps lay the foundation for our code
orgnaization.

The only caveat is that I left `cmd` untouched.  I felt like hiding
commands under the lib folder might be confusing.

```
.
├── client
│   ├── client.go
│   └── interface.go
├── cmd
│   ├── asset
│   │   ├── asset.go
│   │   ├── datastore.go
│   │   └── vpc.go
│   ├── env.go
│   ├── org.go
│   └── root.go
├── config
│   └── config.go
├── debug
│   └── debug.go
├── lib
│   ├── asset
│   │   ├── detail.go
│   │   ├── form.go
│   │   ├── table.go
│   │   └── util.go
│   ├── env
│   │   ├── form.go
│   │   └── table.go
│   ├── op
│   │   └── table.go
│   └── org
│       ├── form.go
│       └── table.go
├── main.go
└── ui
    ├── common
    │   ├── style.go
    │   ├── table.go
    │   └── view.go
    ├── fetch
    │   └── fetch.go
    ├── form
    │   ├── form.go
    │   └── proto.go
    └── loader
        └── loader.go
```

### CODE REVIEW

File: cmd/asset/asset.go [hunk 1]
```diff
@@ -49,26 +49,26 @@ func describeAsset() common.CobraRunE {
...
 }
 
-func assetDescribeRun() common.CobraRunE {
+func assetDescribeRun() config.CobraRunE {
 	return describeAsset()
 }
 
-// destroyAsset - commonly aliased func but also can destroy assets on top level (rds/vpc for example use this)
-func destroyAsset() common.CobraRunE {
+// destroyAsset - configly aliased func but also can destroy assets on top level (rds/vpc for example use this)
```
Comment (madhuravius) [inline cmd/asset/asset.go:62]: typo 🤔

File: lib/asset/table.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	case []cac.AssetOutput:
+		for _, asset := range data {
+			rows = append(rows, generateRowFromData(asset))
+		}
+	case *cac.AssetOutput:
+		rows = append(rows, generateRowFromData(*data))
+	}
+
+	return table.New([]table.Column{
+		table.NewColumn("id", "Id", 40).WithStyle(common.DefaultRowStyle()),
```
Comment (madhuravius) [inline lib/asset/table.go:57]: we should leave todos around some of these if they're mostly the same, we should pipe this to a central utility that generates a table layout for us with a set of column names

**Review by madhuravius — APPROVED** [review]
lgtm, reorg is exactly as planned, minor typos may wanna loop back on

**Comment by tedivm** [comment]
I saw the title of this PR and thought it was on Cloud-API for a minute and you have no idea how excited I got.

**Pull request merged**